cds/modules/announcements/models.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2017 CERN.
+#
+# Invenio is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""CDS Admin models."""
+
+from datetime import datetime
+
+from sqlalchemy import and_, or_
+from sqlalchemy_utils.models import Timestamp
+
+from invenio_db import db
+
+
+class Announcement(db.Model, Timestamp):
+    """Defines a message to show to users."""
+
+    __tablename__ = 'announcements'
+
+    id = db.Column(db.Integer, primary_key=True)
+
+    message = db.Column(db.Text, nullable=False)
+    """The message content."""
+
+    path = db.Column(db.String(100), nullable=True)
+    """Define in which /path the message will be visible."""
+
+    style = db.Column(db.String(20), nullable=False)
+    """Style of the message, for distinguishing warning or info messages."""
+
+    start_date = db.Column(db.DateTime, nullable=False, default=datetime.now())
+    """Start date and time, can be immediate or delayed."""
+
+    end_date = db.Column(db.DateTime, nullable=True)
+    """End date and time, must be after `start` or forever."""
+
+    active = db.Column(db.Boolean('active'), nullable=False, default=True)
```

Comment:
Maybe we could remove this from the database and make a property, i.e. if withing the time range then it is active.

Reply:
This makes me think is we should check, on creation time, if there is already an announcement within the same time range with the same path. (probably we don't need it)

Reply:
About the `active` field: the idea is that if you don't want to "timebox" an announcement, but keep just turn it on/off when you need, then you use the active flag... otherwise you always need a start and end date. But we can do this if we want!

And about the second comment, yes, at beginning I started to think that we could have a piece of code to check if there were overlapping announcements (check for start and end date, but also for path, because you could have 2 announcements at the same time but with different paths), but it was a bit complex maybe, and error prone, so I went for KISS and let humans to ensure that are no or correct overlappings.
Again, we can work on this, but need a little bit of time and test.